### DISCUSSION TITLE
Glyph3DMapper Support - Help with implementation

### DISCUSSION THREAD

**Discussion by LunarLanding** [pyvista/pyvista#1533] [Q&A]
Hello, I am using Glyph3DMapper for performance reasons. 
The panel backend implemented support for it on my request, and now I tried to use my wrapper (just adapted the add_mesh method on the Plotter class) to save a screenshot using the 'static' pyvista backend.
However colors set via scalars are not showing up.
Here's a notebook that contains my wrapper (addGlyph3D), the conda package versions of relevant software, and a demonstration of the issue. If vtk is used directly without setting table values, color shows up both via panel and 'static'.
Notebook: https://gist.github.com/LunarLanding/0e7ca31eec1342fe534e887c05670e80
Screenshot:
![image](https://user-images.githubusercontent.com/4441338/126370063-734eb484-7778-469c-8d14-2a0eec76b115.png)

**Comment by LunarLanding** [comment]
The problem dissapears if I set `interpolate_before_map=False` . WIP to find out why.

**Reply by MatthewFlamm** [reply]
It looks like you are plotting vector values.  This is quite a long code example so it will take a while to unpack for me.  I wonder if this is related to https://github.com/pyvista/pyvista/issues/1420 where mapping using vector data wasn't working when I changed the way pyvista mapped vector data.  See https://github.com/pyvista/pyvista/pull/1387 for the PR that changed the vector data handling for plotting.

**Reply by LunarLanding** [reply]
@MatthewFlamm I'm plotting glyphs using the glyph3D mapper which runs on the GPU. The scalars are 1d, one integer per point (i.e. per glyph instance) in the polydata. I have since whittled down the MVE to less than 80 lines [here](https://gist.github.com/LunarLanding/20697b3bb682570f63c597a5411fedc1) . I think at this point the issue is with vtk and not pyvista, but any pointers would be appreciated.

I think interpolate_before_map=True does not make sense for what I am doing, which is for a given input mesh `source` (a cone mesh in my MVE), for each point P in the input mesh `poly`, make a glyph colored with P's color. Maybe what is happening is that interpolate_before_map does not take into account `poly`'s scalars, and instead takes the colors defined in `source` (which are just the default white for each vertex), making all glyphs white. `vtk.js` implementation, which shows in the first MVE the 'right' colors when interpolate_before_map=True, must be doing it differently or getting different input than the static backend.

**Reply by MatthewFlamm** [reply]
Your analysis seems likely to me.

It seems that `vtk.js` indeed does some manipulation, but I am not an expert in js code to tell for sure. This `colorArray` attribute seems to be a `vtk.js` specific thing by my reading.

https://github.com/Kitware/vtk-js/blob/94f9be2ac7e3301ac2faaa806f473e6a9518dd04/Sources/Rendering/Core/Glyph3DMapper/index.js#L274
